src/tm.h

```diff
@@ -32,7 +32,7 @@ CompressionResultsData tm(
     size_t states,
     size_t alphabet_size,
     unsigned int num_iterations,
-    unsigned int k,
+    std::vector <unsigned int> kvector,
```

Comment:
Codacy is right here, these vectors should be passed by const reference.

This would also make translation to a C API less trivial because of the `std::vector`, but this isn't important for the time being.

Reply:
- Passed variable as reference

Reply:
It's still passed by copy in this line.

```suggestion
    const std::vector <unsigned int>& kvector,
```